vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/util.go

```diff
@@ -39,3 +43,16 @@ func GetAddAnnotationPatch(annotationName, annotationValue string) resource_admi
 		Value: annotationValue,
 	}
 }
+
+// EqPatch returns true if patches are equal by comparing their
+// marshalling result.
+func EqPatch(a, b resource_admission.PatchRecord) bool {
+	aJson, aErr := json.Marshal(a)
+	bJson, bErr := json.Marshal(b)
+	return string(aJson) == string(bJson) && aErr == bErr
+}
+
+// AssertEqPatch asserts patches are equal.
+func AssertEqPatch(t *testing.T, got, want resource_admission.PatchRecord) {
```

Comment:
nit: When golang package is vendored, the _test.go are omitted. IMHO it is good to keep code referencing "testing" separated from main code. I'm not sure what is golang official best practice for this case, though.

Reply:
If we consider keeping "AssertEqPatch" in util, maybe we should also have "assertPatchOneOf" there?

Reply:
Moved to test_utils file. I will move assertPatchOneOf with the next PR (resource updates calculator test)